lib/puppet-lint/plugins/check_parameter_documentation.rb

```diff
@@ -50,7 +52,7 @@ def check
         }
       end
 
-      unless is_private
+      unless is_private and not check_private
```

Comment:
Ruby uses `&&` instead of `and` (same for `||` vs `or`)
```suggestion
      unless is_private && !check_private
```